### ISSUE TITLE
Compilation troubles

### ISSUE THREAD

**Issue by jennyfothergill** [glotzerlab/freud#527]
**Describe the bug**
A clear and concise description of what the bug is and what you expected to happen.
gcc fails with exit code 1 when I run `python setup.py install`

**To Reproduce**
Follow from [source installation instructions](https://freud.readthedocs.io/en/stable/installation.html)

**Error output**
[err_msg.txt](https://github.com/glotzerlab/freud/files/3731149/err_msg.txt)
I've attached all the std err, but here's a snippet that seems important:
```
freud/box.cpp:1606:3: error: ‘PeriodicBuffer’ in namespace ‘freud::box’ does not name a type
   freud::box::PeriodicBuffer *thisptr;
   ^
freud/box.cpp: In function ‘int __pyx_pf_5freud_3box_14PeriodicBuffer___cinit__(__pyx_obj_5freud_3box_PeriodicBuffer*)’:
freud/box.cpp:11534:17: error: ‘struct __pyx_obj_5freud_3box_PeriodicBuffer’ has no member named ‘thisptr’
   __pyx_v_self->thisptr = new freud::box::PeriodicBuffer();
                 ^
freud/box.cpp:11534:31: error: expected type-specifier
   __pyx_v_self->thisptr = new freud::box::PeriodicBuffer();
                               ^
freud/box.cpp:11534:31: error: expected ‘;’
freud/box.cpp: In function ‘void __pyx_pf_5freud_3box_14PeriodicBuffer_4__dealloc__(__pyx_obj_5freud_3box_PeriodicBuffer*)’:
freud/box.cpp:11615:24: error: ‘struct __pyx_obj_5freud_3box_PeriodicBuffer’ has no member named ‘thisptr’
   delete __pyx_v_self->thisptr;
```

**System configuration (please complete the following information):**
 - OS: CentOS 7 cluster
 - Version of Python 3.6.7
 - Version of freud 416f5fbecaa5e26b5e6df8c60013a0292ee72935

**Additional context**

**Comment by jennyfothergill** [comment]
Also, I was able to compile this on my local machine (OSX 10.14) by following the same steps.

**Comment by bdice** [comment]
@jennyfothergill I saw the same issue with @b-butler yesterday. Try adding the flag `--ENABLE-CYTHON`. The files in `freud/*.cpp` are generated by Cython, and they’re currently out of date. (We made changes to the Cython API since we last committed them.) We include these generated files to make it easier to build for users without Cython installed, but with the beta of freud v2, we haven’t kept them up to date with every commit (partly because they’re large). If the flag doesn’t work, try removing the generated C++ files to force it to re-Cythonize.
